src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs

```diff
@@ -90,101 +90,65 @@ public bool ExecuteCommand(ValueTrackingEditorCommandArgs args, CommandExecution
 
             _threadingContext.JoinableTaskFactory.RunAsync(async () =>
                 {
-                    var selectedSymbol = await GetSelectedSymbolAsync(textSpan, document, cancellationToken).ConfigureAwait(false);
-                    if (selectedSymbol is null)
+                    var service = document.Project.Solution.Workspace.Services.GetRequiredService<IValueTrackingService>();
+                    var items = await service.TrackValueSourceAsync(textSpan, document, cancellationToken).ConfigureAwait(false);
+                    if (items.Length == 0)
                     {
-                        // TODO: Show error dialog
                         return;
                     }
 
-                    var syntaxTree = document.GetRequiredSyntaxTreeSynchronously(cancellationToken);
-                    var location = Location.Create(syntaxTree, textSpan);
-
-                    await ShowToolWindowAsync(args.TextView, selectedSymbol, location, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+                    await ShowToolWindowAsync(args.TextView, document, items, cancellationToken).ConfigureAwait(false);
                 });
 
             return true;
         }
 
-        private static async Task<ISymbol?> GetSelectedSymbolAsync(TextSpan textSpan, Document document, CancellationToken cancellationToken)
-        {
-            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var selectedNode = root.FindNode(textSpan);
-            if (selectedNode is null)
-            {
-                return null;
-            }
-
-            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
oh thank goodness.  so now nothign symbolic happens on the client side right?  it's all on the server?